quickwit-doc-mapper/src/query_builder.rs

```diff
@@ -65,8 +74,27 @@ fn has_range_clause(user_input_ast: UserInputAst) -> bool {
             }
             false
         }
-        UserInputAst::Boost(ast, _) => has_range_clause(*ast),
-        UserInputAst::Leaf(leaf) => matches!(*leaf, UserInputLeaf::Range { .. }),
+        UserInputAst::Boost(ast, _) => has_range_clause(ast),
+        UserInputAst::Leaf(leaf) => matches!(**leaf, UserInputLeaf::Range { .. }),
+    }
+}
+
+fn has_search_field(user_input_ast: &UserInputAst) -> bool {
```

Comment:
We should reject queries with the same error message if a query contains a single literal with no field.

In this PR, the query `hello` is rejected.
But the query `hello AND my_field:toto` is not rejected.

We want to reject both with this error message.

To fix it, you just need to change the code for `Clause` below, and make the logic stronger.  

```
 for (_, sub_ast) in sub_queries {
                if !has_search_field(sub_ast) {
                    return false;
                }
            }
            true
```

Reply:
And probably rename the function `has_search_field_everywhere` or a better name you can think of

Reply:
a unit test for this case would be good to :)

Reply:
No sure what you meant by "make the logic stronger" but I refactored and renamed from `has_search_field` to `needs_default_search_field`

Reply:
Here by "stronger", I meant it as make "the predicate stricter".